osu.Game.Rulesets.Osu/Skinning/Legacy/OsuLegacySkinTransformer.cs

```diff
@@ -29,7 +29,7 @@ public OsuLegacySkinTransformer(ISkinSource source)
 
         private void sourceChanged()
         {
-            hasHitCircle = new Lazy<bool>(() => Source.GetTexture("hitcircle") != null);
+            hasHitCircle = new Lazy<bool>(() => FindProvider(s => s.GetTexture("hitcircle") != null) != null);
```

Comment:
I've just done this the simplest way to keep this PR small. The path forward would be to store the `ISkin` here, rather than a `bool`, then passing that in to constructors like `LegacyMainCirclePiece` so they can use it in place of `ISkinSource` for their lookups.